node/src/components/consensus/highway_core/state/index_panorama.rs

```diff
@@ -38,15 +39,15 @@ impl IndexPanorama {
         state: &'a State<C>,
     ) -> Self {
         let mut validator_map: ValidatorMap<IndexObservation> =
-            ValidatorMap::from(vec![IndexObservation::None; panorama.len()]);
+            ValidatorMap::from(vec![IndexObservation::Count(0); panorama.len()]);
         for (vid, obs) in panorama.enumerate() {
             let index_obs = match obs {
-                Observation::None => IndexObservation::None,
-                Observation::Correct(hash) => state
-                    .maybe_unit(hash)
-                    .map_or(IndexObservation::None, |unit| {
-                        IndexObservation::Correct(unit.seq_number)
-                    }),
+                Observation::None => IndexObservation::Count(0),
+                Observation::Correct(hash) => IndexObservation::Count(
+                    state
+                        .maybe_unit(hash)
+                        .map_or(0, |unit| unit.seq_number.saturating_add(1)),
```

Comment:
We add here, but when we log this later, we subtract 1?

Reply:
Yes, `Correct(hash)` means the latest unit we have is `hash`. That means that unit's sequence number plus one is the total number of units we have from that validator.